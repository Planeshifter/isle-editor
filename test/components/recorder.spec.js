// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Recorder from '@isle-project/components/recorder/main.js';


// TESTS //

describe( '<Recorder />', function test() {
	it( 'renders a recorder', () => {
		const recorder = renderer.create(
			<Recorder />
		).toJSON();
		expect( recorder ).toMatchSnapshot();
	});
});
