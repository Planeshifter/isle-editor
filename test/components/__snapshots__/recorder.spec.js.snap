// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recorder /> renders a recorder 1`] = `
<div
  className="recorder-container unselectable"
  style={Object {}}
>
  <div
    className="recorder-rec"
    style={
      Object {
        "color": "rgb(100,100,100)",
      }
    }
  >
    REC
  </div>
  <div
    className="recorder-buttongroup btn-group btn-group-sm btn-group-toggle"
    role="group"
  >
    <label
      className="recorder-togglebutton disabled btn btn-light"
      disabled={false}
    >
      <input
        autoComplete="off"
        checked={false}
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value="screen"
      />
      screen
    </label>
    <label
      className="recorder-togglebutton disabled btn btn-light"
      disabled={false}
    >
      <input
        autoComplete="off"
        checked={false}
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value="camera"
      />
      cam
    </label>
    <label
      className="recorder-togglebutton disabled btn btn-light"
      disabled={false}
    >
      <input
        autoComplete="off"
        checked={false}
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value="audio"
      />
      audio
    </label>
  </div>
  <div
    style={
      Object {
        "background": "rgba(0, 80,255,0.5)",
      }
    }
  />
</div>
`;
